### ISSUE TITLE
Support `Literal` type hint

### ISSUE THREAD

**Issue by rebcabin** [lcompilers/lpython#1939]
The following should not raise an Error in LPython:

```python
└─(10:11:14 on brian-lasr ✹ ✭)──> ~/CLionProjects/lpython/src/bin/lpython lasr_lexer.py                                                               2 ↵ ──(Sun,Jun18)─┘
semantic error: Unsupported type annotation: Literal
  --> lasr_lexer.py:42:12
   |
42 | SEEK_SET : Literal[i32(0)] = i32(0)
   |            ^^^^^^^^^^^^^^^ 
```

It's information to the compiler that the type of `SEEK_SET` is a literal constant. It's kind of an avanced dependent type, and a way to get some semblance of immutability. In any event, it's necessary for an implementation of `Seekable`. 

See https://docs.python.org/3.10/library/typing.html?highlight=literal#typing.Literal and https://docs.python.org/3.10/library/io.html#io.SEEK_SET .
